src/components/Header.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+class Header extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            showFavs: false
+        }
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    handleClick(e) {
+        this.setState({
+            showFavs: !this.state.showFavs
+        }, () => this.props.receiveFavListState(this.state.showFavs))
```

Comment:
I would take the receiver call out of callback and pass to it a pre-calcualated value